[CS2113-T14-2] BagPacker

shannentan opened the pull request:
BagPacker is an app to help travelers manage their packing lists.

DavidVin357 commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,115 @@
+
+When using the `add` function, we have decided to implement a passive function that checks whether the item with the same name already exists in the packingList.
+
+This is done through the `contains()` method in class `PackingList()`, which is called during `execute` in an `AddCommand` object.
```
Maybe class diagram would better clarify interactions between the classes rather than text description?

DavidVin357 commented on docs/img_1.png:
Nice decision to include the reference frame!

DavidVin357 commented on docs/img_1.png:
The `getDescription` calls get pretty confusing here - maybe try to create another activation bar for detaching them to clarify the functionality?

DavidVin357 commented on docs/DeveloperGuide.md:
````diff
@@ -1,29 +1,115 @@
+
+Delete command is used to delete an item from the packing list.
+
+Mechanism: ```DeleteCommand.execute()``` calls the ```PackingList.deleteItem()``` method from the ```PackingList``` class which executes the ```ArrayList.remove()``` method to remove the item from the ```PackingList``` ArrayList.
````
Would be great to see the diagrams here!

SebastianSoewanto replied:
Nice and concise explanation of mechanism

varnika1402 commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,115 @@
+![img_1.png](img_1.png)
+When `execute()` is called in `a`(object of class `AddCommand`), the `addItem(item)` method is called in the object `packingList`. This method will see if method `contains()` will return `true`.
+
+The `contains()` method, which is shown as a reference on the right of the UML diagram, is a boolean method that loops through all items in the packingList, and returns `true` if any of the existing items have the same description as the item to be added in `toAdd`, and `false` otherwise.
+
+When `contains()` returns `true`, method `addItem(item)` will be interrupted and a `DuplicateItemException` will be thrown from `packingList`, which will be caught by `a`. If `contains()` returns false, the item will be added onto `packingList`.
+
+In both scenarios, `ui.printToUser` will be called to print a message to the user. In the former case, `ExistItemError` will be printed, while `AddSuccess` will be shown if the item was added with no issues.
+
```
good workflow explanation of the command

EthanYidong replied:
Would it be better to place this explanation in front of the diagram?

DavidVin357 commented on docs/img_1.png:
Does `Exception thrown back` statement mean actual `throw` or just showing the error to the user? Would be nice to have it more clearly stated or visualized!

DavidVin357 commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,115 @@
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ... | I want to ...                                  | So that I can ...                                                            |
```
Good job with specifying the user stories by version! Gives a good understanding of the actual program and the progress made

varnika1402 commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,115 @@
+| v2b     | user     | Save my packing list                           | keep track of my packing list even after leaving the app                     |
 
 ## Non-Functional Requirements
 
```
should be better to add in instructions for manual testing to get an overview of all commands?

varnika1402 commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,115 @@
+original source as well}
 
 ## Design & implementation
 
```
Should be better to mention the main components of the app architecture to understand the overview of components used?

DavidVin357 commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,115 @@
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
 
 ## Design & implementation
```
One more idea would be to create a detailed class diagram of the Exception classes - this would greatly help developers with what to use when handling new exceptions for new functionality

Gan868611 replied:
Would be good to include Main architectural design and class diagram for each class to showcase the whole project design

varnika1402 commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,115 @@
+This is done through the `contains()` method in class `PackingList()`, which is called during `execute` in an `AddCommand` object.
+
+Below is the UML diagram showing what occurs during `add` function.
+![img_1.png](img_1.png)
```
good use of UML diagram to understand workflow of the command.

EthanYidong replied:
<img width="956" alt="Screenshot 2023-03-30 at 9 10 14 AM" src="https://user-images.githubusercontent.com/50757300/228702831-2438ce39-1af1-4341-87e5-02655b702bfb.png">

Should contains(item) be referenced as an sd block rather than as the condition in the alt block?

EthanYidong replied:
<img width="956" alt="Screenshot 2023-03-30 at 9 10 14 AM" src="https://user-images.githubusercontent.com/50757300/228702900-19df6f3a-3ae7-4bef-8775-370ba1660ecd.png">

Should there be an additional activation bar when packingList calls its add method?

EthanYidong replied:
<img width="956" alt="Screenshot 2023-03-30 at 9 10 14 AM" src="https://user-images.githubusercontent.com/50757300/228702962-b7fb3119-cbeb-494f-ac70-eee741358f40.png">

Should the calls to ui be pointed at its lifeline rather than the entity, and be aligned to its activation bar? And, shouldn't the alt block contain the two possibilities of calls to the UI entity?

xiaoge26 replied:
![image](https://user-images.githubusercontent.com/88525730/228739783-d7c8c630-0bd0-46f3-ac5a-4b572c311d97.png)
Maybe there should be two Ui entities instead of one, as the two possible calls does not call the same Ui object.

Gan868611 replied:
![image](https://user-images.githubusercontent.com/56477330/228739396-364d5c8e-82a1-4da5-b99a-c4cfb95b770a.png)
1) Exception handling in Sequence Diagram is ambiguous. 
2) Missing self invocation activation bar for `.add(item)`
3) Suggest to check out the correct method to showcase static method.(It should point to activation bar)
4) Incorrect use of Reference frame, should check out module website( should have ref and sd frame)
5) Return of `true` and `false` should be dashed line.
6) As it is already in loop frame, there is no need to do self-invocate the whole activation bar.

varnika1402 reviewed (commented):
Overall good explanation of some commands with UML diagram. It would be better to include more detail in the DG.

DavidVin357 reviewed (commented):
The existing sequence diagrams are good enough, but the DG definitely lacks class diagrams. 

Otherwise, the description gives enough information to understand the product and its implementation, though more diagrams would decrease mental effort needed.

EthanYidong commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,115 @@
+
+Below is the UML diagram showing what occurs during `add` function.
+![img_1.png](img_1.png)
+When `execute()` is called in `a`(object of class `AddCommand`), the `addItem(item)` method is called in the object `packingList`. This method will see if method `contains()` will return `true`.
```
Would it be a good idea to give the `AddCommand` instance a more descriptive name?

EthanYidong commented on docs/DeveloperGuide.md:
````diff
@@ -1,29 +1,115 @@
+#### Bye Command
+```ByeCommand``` is used to exit the BagPacker application.
+
+Mechanism: ```ByeCommand.execute()``` updates the static boolean ```isBagPackerRunning``` to be false. 
````
In which class is the `isBagPackerRunning` static variable stored? Would it be a good idea to specify?

SebastianSoewanto commented on docs/DeveloperGuide.md:
````diff
@@ -1,29 +1,115 @@
+2. Create command object - The ```Parser``` class creates a corresponding command object of the relevant command
+3. Execute command object - ```runBagPacker()``` method executes the ```.execute()``` method (overridden by child classes) of the command object 
+   which runs the actual command function
+
````
Would be nice to see some class and sequence diagrams

SebastianSoewanto commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,115 @@
+| v2a     | user     | find an item by name                           | find the pack status of an item without having to go through the entire list |
+| v2a     | user     | remove my packing list                         | clear my list once I am done packing                                         |
+| v2a     | user     | Specify the quantity of an item I need to pack | keep track of individual item quantities being packed                        |
+| v2b     | user     | Save my packing list                           | keep track of my packing list even after leaving the app                     |
```
very well-managed by specifying the version to be implemented and prioritizing the key features.

SebastianSoewanto commented on docs/DeveloperGuide.md:
````diff
@@ -1,29 +1,115 @@
+#### Help Command
+Help command is used to exit the BagPacker application.
+
+Execute: ```HelpCommand.execute()``` prints the following help message.
````
Would be nice to know what type of help commands are available.

Gan868611 commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,115 @@
+[DeveloperGuide.md](DeveloperGuide.md)
 # Developer Guide
 
```
Would be good to include table of contents in the top of the DG

Gan868611 commented on docs/DeveloperGuide.md:
````diff
@@ -1,29 +1,115 @@
+It then updates the ```quantity``` variable according to the quantity inputted by the user.
+
+
+#### Preventing duplicate items
````
`Preventing duplicate items` and `add command` now are on the same layer which makes it ambiguous. Since `preventing duplicate items` is under `add command` then it should be one layer down

Gan868611 commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,115 @@
+
+This is done through the `contains()` method in class `PackingList()`, which is called during `execute` in an `AddCommand` object.
+
+Below is the UML diagram showing what occurs during `add` function.
```
I think specifying the below diagram as **Sequence Diagram** would be a good idea

xiaoge26 replied:
Maybe it is better to add the return of control in your diagram.

xiaoge26 commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,115 @@
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
```
It may be better to give a brief overview of how your application works ] at the beginning of the docs, including the main commands and components, before diving into each component individually.

Gan868611 reviewed (commented):
Should add detailed diagram to all the features

xiaoge26 commented on docs/DeveloperGuide.md:
```diff
@@ -1,29 +1,115 @@
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Command Mechanisms:
+For all valid commands, the mechanism of implementation are as follows:
```
It would be better if you could try standardising whether to add the full stop at the end of the sentence.

xiaoge26 reviewed (commented):
Added my reviews! Good effort!

tangphi committed:
Remove redundant code from merge conflict

tangphi committed:
Update UG with Mac Prompts

tangphi committed:
Resolve merge conflict

tangphi committed:
Merge pull request #125 from tangphi/master

Add Mac prompts to UG

coregano committed:
Revert "Update developer guide"

This reverts commit 961ae6c060d83513eccbc8c4089cb65a817e9961.

tangphi committed:
Edit help message to be more informative

coregano committed:
Merge

tangphi committed:
Update UG for list feature example output

tangphi committed:
Merge pull request #126 from tangphi/master

Edit code formatting and wording

tangphi committed:
Change error message

tangphi committed:
Merge pull request #127 from tangphi/master

Change error message

coregano committed:
Merge branch 'master' of https://github.com/coregano/tp

coregano committed:
Merge branch 'master' of https://github.com/coregano/tp

shannentan committed:
Update User Guide
Resolve GitHub issues #90, #93, #95, #99, #120, #123

tangphi committed:
Create packall feature

tangphi committed:
Merge pull request #132 from tangphi/pack_all_soph

Create packall feature

coregano committed:
feature - automatic save and load added for users

coregano committed:
Merge branch 'master' into save_load

coregano committed:
Merge pull request #133 from coregano/save_load

Save and load feature added

coregano committed:
fix - improve load() function in Storage class to read quantities of above 9

coregano committed:
Merge pull request #134 from coregano/master

fix - improve load() function in Storage class to read quantities of above 9

shannentan committed:
Merge branch 'master' into fix-user-guide-bugs

shannentan committed:
Merge pull request #129 from shannentan/fix-user-guide-bugs

Update User Guide

linuspuah committed:
Update the add function to deal with repeated items of same name

linuspuah committed:
Refactor code to fit checkstyle

linuspuah committed:
Merge branch 'Add_cmd_update'

linuspuah committed:
Update parser class J-unit test

linuspuah committed:
Merge branch 'Junit_parser_v2'

linuspuah committed:
Merge pull request #137 from linuspuah/master

Update the add function to deal with repeated entries, and J-unit for parser class

linuspuah committed:
Update DG for add command

linuspuah committed:
Merge pull request #138 from linuspuah/master

Update DG for add command

coregano committed:
feature - find command added for users to search for items containing the keyword

linuspuah committed:
Add All Command Class J-Unit tests

linuspuah committed:
Merge branch 'master' of https://github.com/linuspuah/tp

linuspuah committed:
Delete HTML reports

linuspuah committed:
Update runtest.bat expected output

linuspuah committed:
Update Unix expected test output

linuspuah committed:
Empty input and expected text ui testing

linuspuah committed:
Revert "Empty input and expected text ui testing"

This reverts commit 3f23dd3b665ede5b94754e1be85949dd8e70e19a.

coregano committed:
fix - code violations removed to pass checkstyle

coregano committed:
Merge pull request #142 from coregano/master

feature - find command added

tangphi committed:
Update packall

tangphi committed:
Merge pull request #143 from tangphi/master

Update packall

tangphi committed:
Work on [PE-D][Tester C] Suggestions for UG #117

tangphi committed:
Merge pull request #144 from tangphi/master

Work on [PE-D][Tester C] Suggestions for UG #117

coregano committed:
Merge pull request #140 from linuspuah/master

Add J-Unit tests for all command classes

coregano committed:
IO fix test

coregano committed:
Merge pull request #145 from coregano/master

Edit EXPECTED-UNIX.TXT to pass IO tests

shannentan committed:
Add feature EditQuantity to edit the total quantity of an item

linuspuah committed:
Merge branch 'master' into feature-edit-total-quantity

linuspuah committed:
Merge pull request #146 from shannentan/feature-edit-total-quantity

Add feature EditQuantity to edit the total quantity of an item

linuspuah committed:
Merge branch 'master' of https://github.com/linuspuah/tp

linuspuah committed:
Refactor Parser and PackingList

linuspuah committed:
Add J-unit tests for new commands

linuspuah committed:
Add ListUnpacked Function

coregano committed:
Merge pull request #151 from linuspuah/master

Add listunpacked command and update some j-unit tests

linuspuah committed:
Fix add item bug

tangphi committed:
Create unpackall command feature

tangphi committed:
Merge pull request #153 from tangphi/master

Create unpackall command feature

linuspuah committed:
Edit UG to include listunpacked and the max integer value supported

linuspuah committed:
Update UG

linuspuah committed:
Merge branch 'master' of https://github.com/linuspuah/tp

linuspuah committed:
Update help messages

linuspuah committed:
Update user guide

linuspuah committed:
Merge pull request #154 from linuspuah/master

Fix bugs and update UG

linuspuah committed:
Delete previous md file

linuspuah committed:
Add plantuml diagram for overarching app

linuspuah committed:
Add linus' PPP

coregano committed:
feature - Storage J-Unit test completed

coregano committed:
fix - imports for storage test

coregano committed:
fix - bug fix

coregano committed:
Merge pull request #157 from coregano/master

feature - storage j-unit test add

linuspuah committed:
Add developer guide main class diagram

linuspuah committed:
Merge branch 'master' of https://github.com/linuspuah/tp

linuspuah committed:
Split Class diagrams into 2 for better readability

linuspuah committed:
Fix bug where wrong error message will be shown when integer overflow

linuspuah committed:
Refactor code to pass checkstyle

linuspuah committed:
Delete old java -jar file

linuspuah committed:
Merge pull request #158 from linuspuah/master

Update DG with overall class diagram and fix bug, add individual PPP

coregano committed:
docs - add Javadocs for methods in ListUnpacked, parser and item

coregano committed:
Merge branch 'master' into master

coregano committed:
Merge pull request #159 from coregano/master

docs - add Javadocs for methods in ListUnpacked, parser and item

tangphi committed:
Add author tags for a segment of test code for add command

tangphi committed:
Create and edit PPP

tangphi committed:
Update UG for unpackall

tangphi committed:
Add JUnit tests for packall and unpackall commands

tangphi committed:
Merge pull request #161 from tangphi/unpackall_ug_dg

Update UG and tests for Unpackall

linuspuah committed:
Add all JavaDocs for classes in commands package

linuspuah committed:
Add JavaDocs for Parser Class

linuspuah committed:
Merge pull request #163 from linuspuah/master

Add JavaDocs for commands package and Parser class

linuspuah committed:
Remove getVariable method

linuspuah committed:
Fix unpackall and packall bugs

coregano committed:
feature - Update runtest.bash

coregano committed:
Merge branch 'master' of https://github.com/coregano/tp

coregano committed:
Merge pull request #164 from coregano/master

Update runtest.bash

linuspuah committed:
Add InvalidQuantityException and fix EditQuantity errors

linuspuah committed:
Fix bugs for pack unpack

linuspuah committed:
Refactor code

linuspuah committed:
Merge pull request #165 from linuspuah/master

Add InvalidQuantityException

shannentan committed:
Update User Guide with information on find command
Add some JUnit Tests in ParserTest

linuspuah committed:
Fix most bugs in Parser Class

linuspuah committed:
Merge pull request #167 from linuspuah/master

Fix most bugs in Parser Class

linuspuah committed:
Merge branch 'master' into master

linuspuah committed:
Merge pull request #166 from shannentan/master

Update User Guide and Add JUnit Tests

linuspuah committed:
Remove disfunctional parser test

linuspuah committed:
Refactor code to pass checkstyle

coregano committed:
feature - update input.txt for io tests
feature - edit developer guide: added special cases for listunpacked, other typo corrections

linuspuah committed:
Merge pull request #168 from linuspuah/master

Refactor code to pass checkstyle

coregano committed:
Merge pull request #169 from coregano/master

feature - update input.txt for io tests

coregano committed:
fix - trial io test

linuspuah committed:
Update J-Unit test to increase coverage

coregano committed:
Merge pull request #170 from coregano/master

fix - trial io test

linuspuah committed:
Merge branch 'AY2223S2-CS2113-T14-2:master' into master

coregano committed:
docs - add packcommand in developer docs

tangphi committed:
Update PPP

tangphi committed:
Update PPP documentation

tangphi committed:
Merge pull request #172 from tangphi/master

PPP

tangphi committed:
Edit formating of PPP

tangphi committed:
Edit PPP spacing

tangphi committed:
Merge pull request #173 from tangphi/master

Edit PPP

linuspuah committed:
Create UML sequence diagram for overall application

linuspuah committed:
Refactor command name in BagPacker Class

linuspuah committed:
Create UML image

linuspuah committed:
Merge branch 'AY2223S2-CS2113-T14-2:master' into master

linuspuah committed:
Merge branch 'master' of https://github.com/linuspuah/tp

shannentan committed:
Update User Guide

shannentan committed:
Update the help message to match the order of the commands in the User Guide

shannentan committed:
Add assert statements

shannentan committed:
Add shannen's PPP

shannentan committed:
Merge branch 'master' into master

shannentan committed:
Merge pull request #174 from shannentan/master

Update User Guide and more

linuspuah committed:
Add Pack and Unpack Command into DG, including Sequence Diagrams

linuspuah committed:
Merge branch 'AY2223S2-CS2113-T14-2:master' into master

linuspuah committed:
Merge branch 'master' of https://github.com/linuspuah/tp

linuspuah committed:
Update Contents page

linuspuah committed:
Merge pull request #176 from linuspuah/master

Update J-Unit and add DG UML diagrams

shannentan committed:
Fix expected outcome code block not displayed correctly for help, list, listunpacked and find
Fix second example input not indented correctly for add and delete

shannentan committed:
Merge pull request #177 from shannentan/master

Fix UG bugs

coregano committed:
Merge branch 'master' of https://github.com/coregano/tp

shannentan committed:
Fix User Guide bugs
Resolve GitHub issues #175

shannentan committed:
Revert "Fix User Guide bugs"

This reverts commit ecc01b0bbc362bc699ace32b1cfa9af939b767d1.

shannentan committed:
Merge branch 'master' of https://github.com/shannentan/tp

linuspuah committed:
Update DG to include Exceptions, and UML diagram for overall command mechanism

shannentan committed:
Fix User Guide bugs
Resolve GitHub issues #175

shannentan committed:
Merge pull request #178 from shannentan/master

Fix User Guide bugs

linuspuah committed:
Merge branch 'master' of https://github.com/linuspuah/tp

linuspuah committed:
Merge pull request #179 from linuspuah/master

Update DG to include Exceptions, and UML diagram for overall command mechanism

coregano committed:
Merge branch 'AY2223S2-CS2113-T14-2:master' into master

coregano committed:
docs - Developer Guide updated

coregano committed:
Merge branch 'master' of https://github.com/coregano/tp

coregano committed:
docs - IncorrectCommand and Command added

coregano committed:
Merge pull request #180 from coregano/master

Docs update

shannentan committed:
Refactor developer guide

shannentan committed:
Rename folder containing diagrams

linuspuah committed:
Merge pull request #181 from shannentan/developer-guide

Refactor developer guide

coregano committed:
docs - storage added in developer docs

coregano committed:
Merge branch 'master' of https://github.com/coregano/tp

# Conflicts:
#	docs/DeveloperGuide.md

linuspuah committed:
Refactor Developer Guide

linuspuah committed:
Merge branch 'master' of https://github.com/linuspuah/tp

# Conflicts:
#	docs/DeveloperGuide.md

linuspuah committed:
Merge pull request #182 from linuspuah/master

Refactor DG

coregano committed:
docs - Add self contribution docs

coregano committed:
Merge branch 'master' of https://github.com/coregano/tp

# Conflicts:
#	docs/DeveloperGuide.md

coregano committed:
Merge pull request #183 from coregano/master

docs - storage added

linuspuah committed:
Add Parser Class to DG

linuspuah committed:
Merge branch 'master' of https://github.com/linuspuah/tp

linuspuah committed:
Refactor DG

linuspuah committed:
Merge pull request #184 from linuspuah/master

Add Parser Class to DG

coregano committed:
docs - instructions for manual testing 50% complete

coregano committed:
Merge remote-tracking branch 'teamRepo/master'

linuspuah committed:
Update Linus' PPP

linuspuah committed:
Merge branch 'AY2223S2-CS2113-T14-2:master' into master

linuspuah committed:
Add UG FAQ's

linuspuah committed:
Merge pull request #185 from linuspuah/master

Add UG FAQ's and update PPP

shannentan committed:
Update PPP

shannentan committed:
Format Developer Guide

coregano committed:
docs - add instructions for manual testing in developer docs
fix - test for io

coregano committed:
Merge branch 'master' of https://github.com/coregano/tp

shannentan committed:
Merge branch 'master' of https://github.com/shannentan/tp

# Conflicts:
#	docs/DeveloperGuide.md

shannentan committed:
Merge pull request #186 from shannentan/master

Update DG

coregano committed:
docs - item for dev guide

coregano committed:
Merge branch 'master' of https://github.com/coregano/tp

# Conflicts:
#	docs/DeveloperGuide.md

linuspuah committed:
Merge pull request #187 from coregano/master

docs - dev guide

linuspuah committed:
Recolour DG and PPP headers

linuspuah committed:
Merge branch 'master' of https://github.com/linuspuah/tp

# Conflicts:
#	docs/DeveloperGuide.md

coregano committed:
fix - delete command test

coregano committed:
io test

shannentan committed:
Update Developer Guide with edit quantity command

linuspuah committed:
Merge pull request #190 from shannentan/master

Update Developer Guide with edit quantity command

linuspuah committed:
Merge branch 'AY2223S2-CS2113-T14-2:master' into master

linuspuah committed:
Add PackAll and UnpackAll commands to DG

linuspuah committed:
Merge branch 'master' of https://github.com/linuspuah/tp

# Conflicts:
#	docs/DeveloperGuide.md

linuspuah committed:
Merge pull request #191 from linuspuah/master

Add PackAll and UnpackAll commands to DG

coregano committed:
Merge branch 'AY2223S2-CS2113-T14-2:master' into master

coregano committed:
merge conflict resolve

shannentan committed:
Update PPP

coregano committed:
Merge pull request #192 from coregano/master

merge conflict resolve

shannentan committed:
Merge pull request #193 from shannentan/master

Update PPP

shannentan committed:
Fix typo in PPP

shannentan committed:
Merge pull request #194 from shannentan/master

Fix typo in PPP

linuspuah committed:
Update Developer guide with packingfunc

linuspuah committed:
Merge branch 'master' of https://github.com/linuspuah/tp

linuspuah committed:
Merge pull request #195 from linuspuah/master

Update Developer guide with packingfunc

tangphi committed:
Update PPP

tangphi committed:
Merge pull request #196 from tangphi/master

Update PPP

coregano committed:
Final PPP upload

tangphi committed:
Update naming of quantities in UG

linuspuah committed:
Refactor code and fix links in UG

linuspuah committed:
Merge branch 'master' of https://github.com/linuspuah/tp

tangphi committed:
Merge pull request #197 from tangphi/master

Update naming of quantities in UG

linuspuah committed:
Merge branch 'master' of https://github.com/linuspuah/tp

# Conflicts:
#	docs/UserGuide.md

linuspuah committed:
Merge pull request #198 from linuspuah/master

Refactor code and fix links in UG

shannentan committed:
Fix DG bugs

shannentan committed:
Merge branch 'master' of https://github.com/shannentan/tp

# Conflicts:
#	docs/DeveloperGuide.md

shannentan committed:
Merge pull request #199 from shannentan/master

Fix DG links

linuspuah committed:
Update PPP

linuspuah committed:
Merge branch 'AY2223S2-CS2113-T14-2:master' into master

shannentan committed:
Fix UG table of contents links

shannentan committed:
Merge pull request #200 from shannentan/master

Fix UG table of contents links

linuspuah committed:
Merge pull request #201 from linuspuah/master

Update Linus' PPP

linuspuah committed:
Refactor Linus' PPP

linuspuah committed:
Merge branch 'AY2223S2-CS2113-T14-2:master' into master

linuspuah committed:
Merge pull request #202 from linuspuah/master

Update PPP

coregano committed:
Merge branch 'AY2223S2-CS2113-T14-2:master' into master

linuspuah committed:
Refactor PPP

linuspuah committed:
Merge branch 'master' of https://github.com/linuspuah/tp

linuspuah committed:
Refactor PPP

linuspuah committed:
Merge branch 'AY2223S2-CS2113-T14-2:master' into master

tangphi committed:
Correct error message output

tangphi committed:
Update README

tangphi committed:
Merge pull request #204 from tangphi/master

Bug fixing

linuspuah committed:
Merge branch 'AY2223S2-CS2113-T14-2:master' into master

shannentan committed:
Fix bug - characters get cut off in pdf
Add line breaks

shannentan committed:
Merge pull request #205 from shannentan/master

Fix bug - characters get cut off in pdf

linuspuah committed:
Merge branch 'AY2223S2-CS2113-T14-2:master' into master

linuspuah committed:
Bug fix unpackall and packall

linuspuah committed:
Merge pull request #203 from linuspuah/master

Refactor Linus' PPP

shannentan committed:
Add line break to PPP

shannentan committed:
Merge branch 'AY2223S2-CS2113-T14-2:master' into master

shannentan committed:
Merge pull request #206 from shannentan/master

Edit PPP

coregano committed:
Merge branch 'AY2223S2-CS2113-T14-2:master' into master

shannentan committed:
Fix DG broken link
Add line break

shannentan committed:
Merge pull request #207 from shannentan/master

Fix DG broken link and more

coregano committed:
Merge branch 'AY2223S2-CS2113-T14-2:master' into master

coregano committed:
code block horizontal scroll fixed

coregano committed:
Merge pull request #208 from coregano/master

code block fix

coregano committed:
handled test fix

coregano committed:
fix

linuspuah committed:
Merge pull request #209 from coregano/master

handled test fix

coregano committed:
io test fix

coregano committed:
Merge pull request #211 from coregano/master

fix io tests
